client/js/compute.js

```diff
@@ -0,0 +1,578 @@
+      desc: 'returns a string with the first occurence of string y replaced by string z',
+      sample: 'var a = ${x} replace ${y} ${z}',
+      call: function(v, x, y, z) { return v = x.replace(y, z) }
+    },
+    {
+      name: 'replaceAll',
+      desc: 'returns a string with all occurences of string y replaced by string z',
+      sample: 'var a = ${x} replaceAll ${y} ${z}',
+      call: function(v, x, y, z) { return v = x.replaceAll(y, z) }
+    },
+    {
+      name: 'substr',
+      desc: 'extracts and returns part of a string, starting at index y for z number of characters - or to the end (if z is zero/omitted)',
+      sample: 'var a = ${x} substr ${y} ${z}',
+      call: function(v, x, y, z) { return v = (z !== undefined) ? x.substr(y, z) : x.substr(y) }
+    },
+    {
+      name: 'getIndex',
+      desc: 'returns index y of a string or array x',
+      sample: 'var a = ${x}.${y}\nvar a = ${x} getIndex ${y}',
+      call: function(v, x, y) { return v = x[y] }
+    },
+    {
+      name: 'setIndex',
+      desc: 'sets the index x of the given variable a to the value y',
+      sample: 'var a.${x} = ${y}\nvar a = setIndex ${x} ${y}',
+      call: function(v, x, y) { v[x] = y; return v }
+    },
+    {
+      name: 'from',
+      desc: 'converts and returns an array of the given string or object',
+      sample: 'var a = from ${x}',
+      call: function(v, x) { return v = Array.from(x) }
+    },
+    {
+      name: 'isArray',
+      desc: 'returns true if the object is an array, and false if not',
+      sample: 'var a = isArray ${x}',
+      call: function(v, x) { return v = Array.isArray(x) }
+    },
+    {
+      name: 'concatArray',
+      desc: 'returns an array containing the elements of the two joined arrays',
+      sample: 'var a = ${x} concatArray ${y}',
+      call: function(v, x, y) { return v = x.concat(y) }
+    },
+    {
+      name: 'pop',
+      desc: 'returns an array after removing its last element',
+      sample: 'var a = ${x} pop',
+      call: function(v, x) { return v = x.pop() }
+    },
+    {
+      name: 'reverse',
+      desc: 'returns an array after reversing the order of its elements',
+      sample: 'var a = ${x} reverse',
+      call: function(v, x) { return v = x.reverse() }
+    },
+    {
+      name: 'shift',
+      desc: 'returns an array after removing its first element',
+      sample: 'var a = ${x} shift',
+      call: function(v, x) { return v = x.shift() }
+    },
+    {
+      name: 'sort',
+      desc: 'returns an array after sorting its elements alphabetically',
+      sample: 'var a = ${x} sort',
+      call: function(v, x) { return v = x.sort() }
+    },
+    {
+      name: 'numericSort',
+      desc: 'returns an array after sorting its elements numerically',
+      sample: 'var a = ${x} numericSort',
+      call: function(v, x) { return v = x.sort((a, b) => a.toString().localeCompare(b, undefined, {numeric: true, ignorePunctuation: true})) }
+    },
+    {
+      name: 'findIndex',
```

Comment:
`y` needs to be a function so it will never work as intended. We should remove it but the fileupdater should emulate its erroneously behavior.

Please change the description though and state that it doesn't work.

Reply:
From my testing, the behavior is that it always returns a zero no matter what.  Any variable, no variable, any array, no array, null variable, zero variable - always returns zero.